packages/core/src/browser/tree/tree-widget.tsx

```diff
@@ -126,6 +162,9 @@ export class TreeWidget extends ReactWidget implements StatefulWidget {
             this.decoratorService,
             this.decoratorService.onDidChangeDecorations(() => this.updateDecorations())
         ]);
+        this.model.onExpansionChanged(() => {
```

Comment:
make sure that all listeners are collected into `this.toDispose`

Reply:
fixed